android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -476,145 +476,176 @@ public void run() {
-			}else{
-				return  null;
+			int motionEventAction = sdlMotionEvent.getMotionEventAction(touchType, touchEvent);
+			long downTime = sdlMotionEvent.downTime;
+			long eventTime = sdlMotionEvent.eventTime;
+			pointer = sdlMotionEvent.getPointerById(touchEvent.getId());
+			if (pointer != null) {
+				pointer.setCoords(touchCoord.getX() * touchScalar[0], touchCoord.getY() * touchScalar[1]);
 			}
-		}
 
-		int eventAction = sdlMotionEvent.getMotionEvent(touchType, pointerProperties);
-		long startTime = sdlMotionEvent.startOfEvent;
+			MotionEvent.PointerProperties[] pointerProperties;
+			MotionEvent.PointerCoords[] pointerCoords;
+			pointerProperties = new MotionEvent.PointerProperties[sdlMotionEvent.pointers.size()];
+			pointerCoords = new MotionEvent.PointerCoords[sdlMotionEvent.pointers.size()];
+
+			for (int i = 0; i < sdlMotionEvent.pointers.size(); i++) {
+				pointerProperties[i] = new MotionEvent.PointerProperties();
+				pointerProperties[i].id = sdlMotionEvent.getPointerByIndex(i).id;
+				pointerProperties[i].toolType = MotionEvent.TOOL_TYPE_FINGER;
+
+				pointerCoords[i] = new MotionEvent.PointerCoords();
+				pointerCoords[i].x = sdlMotionEvent.getPointerByIndex(i).x;
+				pointerCoords[i].y = sdlMotionEvent.getPointerByIndex(i).y;
+				pointerCoords[i].orientation = 0;
+				pointerCoords[i].pressure = 1.0f;
+				pointerCoords[i].size = 1;
+			}
 
-		//If the motion event should be finished we should clear our reference
-		if(eventAction == MotionEvent.ACTION_UP || eventAction == MotionEvent.ACTION_CANCEL){
-			sdlMotionEvent = null;
+			motionEvent = MotionEvent.obtain(downTime, eventTime, motionEventAction,
+					sdlMotionEvent.pointers.size(), pointerProperties, pointerCoords, 0, 0, 1,
+					1, 0, 0, InputDevice.SOURCE_TOUCHSCREEN, 0);
+			motionEventList.add(motionEvent);
+
+			if(motionEventAction == MotionEvent.ACTION_UP || motionEventAction == MotionEvent.ACTION_CANCEL){
+				//If the motion event should be finished we should clear our reference
+				sdlMotionEvent.pointers.clear();
+				sdlMotionEvent = null;
+				break;
+			} else if((motionEventAction & MotionEvent.ACTION_MASK) == MotionEvent.ACTION_POINTER_UP){
+				sdlMotionEvent.removePointerById(touchEvent.getId());
+			}
 		}
 
-		return MotionEvent.obtain(startTime, SystemClock.uptimeMillis(), eventAction, eventListSize, pointerProperties, pointerCoords, 0, 0,1,1,0,0, InputDevice.SOURCE_TOUCHSCREEN,0);
+		return motionEventList;
 	}
 
 	/**
 	 * Keeps track of the current motion event for VPM
 	 */
-	private static class SdlMotionEvent{
-		long startOfEvent;
-		SparseIntArray pointerStatuses = new SparseIntArray();
+	private static class SdlMotionEvent {
+		class Pointer {
+			int id;
+			float x;
+			float y;
+			Pointer (int id) {
+				this.id = id;
+				this.x = 0.0f;
+				this.y = 0.0f;
+			}
+			void setCoords(float x, float y) {
+				this.x = x;
+				this.y = y;
+			}
+		}
+
+		private List<Pointer> pointers = new ArrayList<>();
```

Comment:
This is an unsynchronized list but has multiple methods that can alter its contents which can lead to unexpected results including concurrent modification exceptions. Either all methods accessing this list need to be synchronized to the same lock, or the data type will need to be changed.

Reply:
Makes sense. I will use CopyOnWriteArrayList for the pointers list